### PR TITLE
NCSD1275: Adds simple azure monitor webhook for piping alerts to slack

### PR DESCRIPTION
**Pull request by thelious1974**

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (thelious1974) [inline .vscode/extensions.json]: The .vscode folder contains items that are used when developing the azure function in VSCode

File: MonitorAlert/run.ps1 [hunk 2]
```diff
@@ -0,0 +1,150 @@
+<#
```
Comment (thelious1974) [inline MonitorAlert/run.ps1]: This is the main file of the function.

It essentially does the following steps:
* grabs inputs
* validates inputs
* generates slack message
* posts slack message

File: MonitorAlert/run.ps1 [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        [hashtable] $message
+    )
+
+    $serializedMessage = "payload=$($message | ConvertTo-Json)"
+
+    Invoke-RestMethod -Uri https://hooks.slack.com/services/$($slackToken) -Method POST -UseBasicParsing -Body $serializedMessage
+}
+
+# Parse query and get settings from environment variables
+$channel = $Request.Query.Channel
```
Comment (NickGraham101) [inline MonitorAlert/run.ps1]: Does this need a try catch to throw a useful error if Channel isn't present?

File: MonitorAlert/run.ps1 [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+https://docs.microsoft.com/en-us/azure/azure-monitor/platform/alerts-metric-near-real-time#payload-schema
+
+Some possible improvements:
+    * Update to add support for the common alert schema
+#>
+
+using namespace System.Net
+
+# Input bindings are passed in via param block.
+param($Request, $TriggerMetadata)
```
Comment (NickGraham101) [inline MonitorAlert/run.ps1]: Parameters should be documented seperately

File: MonitorAlert/run.ps1 [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+# Write to the Azure Functions log stream.
+Write-Host "PowerShell HTTP trigger function processed a request."
+
+function EncodeSlackHtmlEntities {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $ToEncode
```
Comment (NickGraham101) [inline MonitorAlert/run.ps1]: Parameters should be documented seperately

File: MonitorAlert/run.ps1 [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        Replace(">", "&gt;")
+
+    return $encoded
+}
+
+function New-SlackMessageFromAlert
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $channel,
```
Comment (NickGraham101) [inline MonitorAlert/run.ps1]: Parameters should be documented seperately

File: MonitorAlert/run.ps1 (hunk 3 above)
Comment (thelious1974) [reply MonitorAlert/run.ps1]: If the channel isn't present,  it returns a null and is caught by some validation further down and returns "BadRequest" http status code to the client with info on how to resolve it

File: MonitorAlert/run.ps1 (hunk 5 above)
Comment (thelious1974) [reply MonitorAlert/run.ps1]: documentation added

File: MonitorAlert/run.ps1 (hunk 6 above)
Comment (thelious1974) [reply MonitorAlert/run.ps1]: documentation added

File: Modules/MonitorAlertHelperModule/MonitorAlertHelperModule.psm1 [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+.Parameter Body
+
+String.  The text to return to the client.
+
+#>
+
+    param(
+        [Parameter(Mandatory=$true)]
+        [HttpStatusCode] $Status,
+        [string] $Body=""
```
Comment (NickGraham101) [inline Modules/MonitorAlertHelperModule/MonitorAlertHelperModule.psm1]: Looks like Body is always specified when this is called.  Do we need to allow a default value?

**Pull request merged**